src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -568,28 +568,23 @@ Node *RegionNode::Ideal(PhaseGVN *phase, bool can_reshape) {
       PhaseIterGVN *igvn = phase->is_IterGVN();
       del_req(i);               // Yank path from self
       del_it = i;
-      uint max = outcnt();
-      DUIterator j;
-      bool progress = true;
-      while(progress) {         // Need to establish property over all users
-        progress = false;
-        for (j = outs(); has_out(j); j++) {
-          Node *n = out(j);
-          if( n->req() != req() && n->is_Phi() ) {
-            assert( n->in(0) == this, "" );
-            igvn->hash_delete(n); // Yank from hash before hacking edges
-            n->set_req_X(i,nullptr,igvn);// Correct DU info
-            n->del_req(i);        // Yank path from Phis
-            if( max != outcnt() ) {
-              progress = true;
-              j = refresh_out_pos(j);
-              max = outcnt();
-            }
-          }
+
+      for (DUIterator_Fast jmax, j = fast_outs(jmax); j < jmax; j++) {
+        Node* use = fast_out(j);
+
+        if(use->req() != req() && use->is_Phi()) {
+          assert(use->in(0) == this, "");
```

Comment:
```suggestion
          assert(use->in(0) == this, "unexpected control input");
```

Reply:
updated. 

Could you help me submit this to your CI validation? TBH, I don't know why the original author starts over the loop. In my understanding, it doesn't delete any phi node but just put them into worklist.

Reply:
I did, all tests passed. I'm also not sure why that loop was there.